File: algorithm/typescript/BOJ2439.ts
// problem : https://www.acmicpc.net/problem/2439

import fs = require("fs");

const N: number = Number(fs.readFileSync("/dev/stdin").toString());
const answer: string[] = new Array();

for (let i = N; i > 0; i--) {
  for (let j = 0; j < i - 1; j++) {
    answer.push(" ");
  }

  for (let k = N; k >= i; k--) {
    answer.push("*");
  }

  answer.push("\n");
}

answer.pop();
console.log(answer.join(""));

File: algorithm/python/BOJ2439.py
# https://www.acmicpc.net/problem/2439

import sys

N = int(sys.stdin.readline().rstrip())
answer = ""

for i in range(N, 0, -1):
    for j in range(0, i - 1):
        answer += " "

    for k in range(N, i - 1, -1):
        answer += "*"

    answer += "\n"

print(answer)
